### PR TITLE
new dockerize app

### PR DESCRIPTION
**Pull request by MrSancho42**
App was dockerized

### CODE REVIEW

**Review by dmytrosmetanin — APPROVED** [review]
approved

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    build:
+      context: .
+      dockerfile: front-end.Dockerfile
+    container_name: front-end
+    ports:
+      - 80:8080
+    depends_on:
+      - api
+
+volumes:
```
Comment (vvelizariy) [inline docker-compose.yaml:41]: Broken indentation?

File: requirements.txt [hunk 2]
```diff
@@ -9,3 +9,4 @@ pymongo==3.12.0
 python-decouple==3.4
 pytz==2021.1
 sqlparse==0.2.4
+gunicorn==20.1.0
```
Comment (vvelizariy) [inline requirements.txt]: requirements should be sorted alphabetically. Just pip freeze(dependencies will be sorted) and paste to file,

File: front-end.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# copy react file to image
+COPY ./front-end/ .
+
+# install all node dependencies from package.json
+RUN npm install
+# build react app
+RUN npm run build
+
+# get nginx image
+FROM nginx:alpine
```
Comment (vvelizariy) [inline front-end.Dockerfile]: I'd prefer installing nginx instead of mixing images via FROM.

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      - "8000:8000"
+    depends_on:
+      - postgres
+
+# TODO mongodb
+
+  front-end:
+    build:
+      context: .
+      dockerfile: front-end.Dockerfile
```
Comment (vvelizariy) [inline docker-compose.yaml]: please don't use - in filenames, more reasonable would be to name it as ui.Dockerfile

File: api.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# get python image
+FROM python:latest
+
+# python output is sent straight to terminal without being buffered 
+ENV PYTHONUNBUFFERED=1
```
Comment (vvelizariy) [inline api.Dockerfile]: Does it gives any visible effect? I think can be omitted.

File: docker-compose.yaml (hunk 1 above)
Comment (MrSancho42) [reply docker-compose.yaml:41]: I don't understand what you mean

File: api.Dockerfile (hunk 5 above)
Comment (MrSancho42) [reply api.Dockerfile]: Tutorials said it don't save logs in container

File: front-end.Dockerfile (hunk 3 above)
Comment (MrSancho42) [reply front-end.Dockerfile]: So I should install nginx in node container and start it without using nginx image?

**Review by Viktorika-p — APPROVED** [review]

File: front-end.Dockerfile (hunk 3 above)
Comment (vvelizariy) [reply front-end.Dockerfile]: Yes

File: docker-compose.yaml (hunk 1 above)
Comment (vvelizariy) [reply docker-compose.yaml:41]: Sorry, missed that it is common volume declaration. Ignore it.

**Review by dmytrosmetanin — APPROVED** [review]
Approved

**Review by DmytroBaryshniuk — APPROVED** [review]

**Pull request merged**